lib/matplotlib/ticker.py

```diff
@@ -416,15 +417,17 @@ class ScalarFormatter(Formatter):
     axes.formatter.limits rc parameter.
 
     """
+    _usingOffset = False
+    _usingScaling = False
 
     def __init__(self, useOffset=None, useMathText=None, useLocale=None):
         # useOffset allows plotting small data ranges with large offsets: for
         # example: [1+1e-9,1+2e-9,1+3e-9] useMathText will render the offset
         # and scientific notation in mathtext
 
-        if useOffset is None:
-            useOffset = rcParams['axes.formatter.useoffset']
-        self.set_useOffset(useOffset)
+        self.set_useOffset(False)
```

Comment:
It still honors the rcparam, see line 671. 

The variabel _useoffset now indicates if the offset has been set by either the user or the algorithm choosing a good offset.
`axes.formatter.useoffset = False` now disables the automatic algorithm. The user can still manually set an offset
`axes.formatter.useoffset = True` allows the automatic algorithm to set an offset. The user can still override this.